e2e/config/config.go

```diff
@@ -71,8 +71,17 @@ type Deployer struct {
 	Type string `json:"type"`
 	// Description is a human-readable description of the deployer.
 	Description string `json:"description"`
+	Recipe      Recipe `json:"recipe,omitempty"`
 }
 
+type Recipe struct {
+	// RecipeName is the name of the recipe to use.
+	// If RecipeName is "generate", the generated recipe will use this name.
+	// If RecipeName is "vm", ramen internal recipe for the vm will be used.
+	RecipeName string `json:"recipeName"`
```

Comment:
Repeating Recipe in the name of field is not needed, so this should be Name. But this is not really a name (any value you like) but an enum (generate, vm). Other possible names: type, mode, strategy.